src/NServiceBus.NHibernate.Tests/TimeoutPersister/When_removing_timeouts_from_the_storage.cs

```diff
@@ -79,30 +81,97 @@ public void Should_remove_timeouts_by_id()
         }
 
         [Test]
-        public void Should_remove_timeouts_by_sagaid()
+        public void TryRemove_should_return_false_when_timeout_already_deleted()
+        {
+            var timeout = new TimeoutData();
+
+            persister.Add(timeout);
+
+            Assert.IsTrue(persister.TryRemove(timeout.Id));
+            Assert.IsFalse(persister.TryRemove(timeout.Id));
+        }
+
+        [Test]
+        public void TryRemove_should_work_with_concurrent_transactions()
+        {
+            var timeout = new TimeoutData()
+            {
+                Id = Guid.NewGuid().ToString(),
```

Comment:
oh that id assignment shouldn't be there. Thats some remainder from the SQLite tests. Going to remove that.